xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  BylineLgView.swift
+//  Subconscious (iOS)
+//
+//  Created by Ben Follington on 29/3/2023.
+//
+
+import Foundation
+import SwiftUI
+
+enum ProfileActionButton {
+    case follow
+    case following
+    case editProfile
+}
+
+
+struct UserProfileHeaderView: View {
+    var user: UserProfile
+    var statistics: UserProfileStatistics?
+    
+    var isFollowingUser: Bool
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: AppTheme.unit3) {
+            HStack(alignment: .center, spacing: AppTheme.unit3) {
+                ProfilePic(image: Image(user.pfp))
+                
+                HStack(alignment: .firstTextBaseline, spacing: AppTheme.unit) {
+                    switch (user.category) {
+                    case .human:
+                        PetnameBylineView(petname: user.petname)
+                    case .geist:
+                        PetnameBylineView(petname: user.petname)
+                    case .you:
+                        PetnameBylineView(petname: user.petname)
+                        Text("(you)")
+                            .foregroundColor(.secondary)
+                    }
+                    
+                }
+                
+                Spacer()
+                
+                if user.category == .you {
+                    Button(action: {}, label: {
+                        Label("Edit Profile", systemImage: "pencil")
+                    })
+                    .buttonStyle(PillButtonStyle(size: .small))
+                    .frame(maxWidth: 160)
+                } else {
+                    if isFollowingUser {
+                        Button(action: {}, label: {
+                            Label("Following", systemImage: "person.fill.checkmark")
+                        })
+                        .buttonStyle(GhostPillButtonStyle(size: .small))
+                        .frame(maxWidth: 160)
+                    } else {
+                        Button(action: {}, label: {
+                            Text("Follow")
+                        })
+                        .buttonStyle(PillButtonStyle(size: .small))
+                        .frame(maxWidth: 160)
+                    }
+                }
```

Comment:
Considering extracting to a function or sub-view because this is ugly. However it's likely placeholder anyway.